lib/ecmascript.ts

```diff
@@ -395,7 +397,7 @@ export function ToObject<T>(value: T): T extends Record<string, unknown> ? T : o
 
 // Adapted from https://github.com/ljharb/es-abstract/blob/main/2022/CopyDataProperties.js
 // but simplified (e.g. removed assertions) for this polyfill to reduce bundle size.
-export function CopyDataProperties<K extends string | symbol, T extends Record<K, unknown>>(
+export function CopyDataProperties<K extends string | symbol | number, T extends Record<K, unknown>>(
```

Comment:
Nit: could `string | symbol | number` also be `PropertyKey` (built-in type for this)?